pyvista/core/dataobject.py

```diff
@@ -492,20 +492,19 @@ def __getstate__(self):
         writer = _vtk.vtkDataSetWriter()
         writer.SetInputDataObject(self)
         writer.SetWriteToOutputString(True)
-        writer.SetFileTypeToASCII()
+        writer.SetFileTypeToBinary()
         writer.Write()
-        to_serialize = writer.GetOutputString()
+        to_serialize = writer.GetOutputStdString()
         state['vtk_serialized'] = to_serialize
         return state
 
     def __setstate__(self, state):
         """Support unpickle."""
         vtk_serialized = state.pop('vtk_serialized')
         self.__dict__.update(state)
-
         reader = _vtk.vtkDataSetReader()
         reader.ReadFromInputStringOn()
-        reader.SetInputString(vtk_serialized)
+        reader.SetBinaryInputString(vtk_serialized, len(vtk_serialized))
```

Comment:
Can we detect whether a non-binary string is in the pickled object, for backwards compatibility?  I'm not sure if these objects are backwards compatible in general, so it may not be worth it.

Reply:
We should be able to check if the type is `str` or `byte`. I guess the use case would be if somebody has pickled objects laying around? This should be backwards compatible.

Reply:
I'm waiting on this to be resolved before reviewing.

Reply:
Pushed this change